protocol/src/main/java/com/amazon/opendistroforelasticsearch/sql/protocol/response/QueryResponse.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *    Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License").
+ *    You may not use this file except in compliance with the License.
+ *    A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    or in the "license" file accompanying this file. This file is distributed
+ *    on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *    express or implied. See the License for the specific language governing
+ *    permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.protocol.response;
+
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprValue;
+import com.amazon.opendistroforelasticsearch.sql.data.model.ExprValueUtils;
+import lombok.RequiredArgsConstructor;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Query response that encapsulates query results and isolate {@link ExprValue} related from formatter implementation.
+ */
+@RequiredArgsConstructor
+public class QueryResponse implements Iterable<Object[]> {
+
+    /**
+     * Results which are collection of expression
+     */
+    private final Collection<ExprValue> exprValues;
+
+
+    /**
+     * @return  size of results
+     */
+    public int size() {
+        return exprValues.size();
+    }
+
+    /**
+     * Parse column name from results
+     * @return  mapping from column names to its expression type
+     */
+    public Map<String, String> columnNameTypes() {
+        if (exprValues.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, ExprValue> tupleValue = getFirstTupleValue();
```

Comment:
it is possible the ExprValue has missing value, in this case, the type value may missing.
e.g. [{"name": "susan"}, {"name": "bob", "age":30}].
it may required the type info from the planner?

Reply:
let's add Todo here.

Reply:
Sure, I'm adding comment and a disabled UT. Thanks!